value-processor/src/org/immutables/value/processor/Immutables.generator

```diff
@@ -702,26 +702,52 @@ public [type.typeAbstract.relative] [v.names.with]([v.atNullability][v.type] val
 
   /**
    * Construct a new immutable {@code [type.name]} instance.
-[for v in type.constructorArguments]
+  [for v in type.constructorArguments]
    * @param [v.name] The value for the {@code [v.name]} attribute
-[/for]
+  [/for]
    * @return An immutable [type.name] instance
    */
   public static [type.generics.spaceAfter][type.typeValue] [type.names.of]([for v in type.constructorArguments][if not for.first], [/if][v.atNullability][eachLine v.constructorParameterInjectedAnnotations][v.type] [v.name][/for]) {
     return [type.names.of]([for v in type.constructorArguments][if not for.first], [/if][if v.requiresAlternativeStrictConstructor]([constructorAcceptType v]) [/if][v.name][/for]);
   }
+
+  [if type.hasOptionalConstructorArguments]
```

Comment:
suggest to remove this variation, I think we don't need to mess with special `type.requiresAlternativeStrictConstructor` mode here (we're inside this condition). Unless this is really needed for something, any examples? The initial idea was that we use in general `Iterable<? extends Something>` in params for collection, but for compatibility (was needed at some point, I guess) we also adding constructor where those are `List<Something>` originally declared in attribute. You can try mixing with constructor arguments of new optional with those generic lists and see if we can get duplicate or otherwise unnecessary constructors. In general I consider these new (which accept nullable for optional) constructors as anything but strict so would prefer to not multiply such combinations.

Reply:
Right, I will remove it, I figured that it should be there for the same reason that the current alternative is there but I have no example. If it would make sense it can be added later.